### ISSUE TITLE
(t) Django management commands are failing to run

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockstor-core#2377]
Upon a fresh source build of Rockstor, running any Django management command fails with the same error as below:
```
buildvm:/opt/rockstor-dev # /opt/rockstor-dev/bin/django showmigrations
Traceback (most recent call last):
  File "/opt/rockstor-dev/bin/django", line 66, in <module>
    main()
  File "/opt/rockstor-dev/bin/django", line 62, in main
    execute_from_command_line(sys.argv)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/core/management/__init__.py", line 308, in execute
    settings.INSTALLED_APPS
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named py
```
After a quick inspection, it seems related to a change in the syntax when setting the `DJANGO_SETTINGS_MODULE` environment variable in #2365. We indeed have:
https://github.com/rockstor/rockstor-core/blob/57998ec6980510f5efef6e3adb2bfb7b2b51d03c/conf/django-hack.py#L46

Using the dotted path syntax instead (as prior to #2365) seems to work:
```python
    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "rockstor.settings")
```

For instance, we can now run:
```shell
$ /opt/rockstor-dev/bin/django showmigrations
/opt/rockstor-dev/eggs/psycopg2-2.7.4-py2.7-linux-x86_64.egg/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.)
admin
 [ ] 0001_initial
 [ ] 0002_logentry_remove_auto_add
auth
 [X] 0001_initial
 [X] 0002_alter_permission_name_max_length
 [X] 0003_alter_user_email_max_length
 [X] 0004_alter_user_username_opts
 [X] 0005_alter_user_last_login_null
 [X] 0006_require_contenttypes_0002
 [X] 0007_alter_validators_add_error_messages
 [X] 0008_alter_user_username_max_length
contenttypes
 [X] 0001_initial
 [X] 0002_remove_content_type_name
oauth2_provider
 [X] 0001_initial
 [X] 0002_08_updates
 [X] 0003_auto_20160316_1503
 [X] 0004_auto_20160525_1623
 [X] 0005_auto_20170514_1141
 [X] 0006_auto_20171214_2232
sessions
 [ ] 0001_initial
sites
 [ ] 0001_initial
 [ ] 0002_alter_domain_unique
smart_manager
 [ ] 0001_initial
 [ ] 0002_auto_20170216_1212
storageadmin
 [X] 0001_initial
 [X] 0002_auto_20161125_0051
 [X] 0003_auto_20170114_1332
 [X] 0004_auto_20170523_1140
 [X] 0005_auto_20180913_0923
 [X] 0006_dcontainerargs
 [X] 0007_auto_20181210_0740
 [X] 0008_auto_20190115_1637
 [X] 0009_auto_20200210_1948
 [X] 0010_sambashare_time_machine
 [X] 0011_auto_20200314_1207
 [X] 0012_auto_20200429_1428
 [X] 0013_auto_20200815_2004
 [X] 0014_rockon_taskid
 [X] 0015_auto_20210513_1830
```

@phillxnet , does that look like a sensible change or am I missing the root cause of the problem here?

**Comment by phillxnet** [comment]
If all goes well and you find we can undo the change in
(t) data-collector (Dashboard backend) fails to start #2368 #2369
the do by all means add that to a focused pr. I can then merge to testing and we have our hack undone and a documented proper fix. I can then re-base here to include those discreet fixes as I stagger on with the unit tests thing.

**Comment by FroggyFlox** [comment]
> If all goes well and you find we can undo the change in
> (t) data-collector (Dashboard backend) fails to start #2368 #2369

Unfortunately, it does not seem to be the case. I have tried the following:
```python
os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings")
```
or
```python
os.environ.setdefault("DJANGO_SETTINGS_MODULE", "rockstor.settings")
```

Both options work when trying to run `/opt/rockstor-dev/bin/django showmigrations`, but if I undo the changes in #2369 (by commenting the `django.setup()` line), then I get the same "Apps aren't loaded" error that was described in #2368.

**Comment by phillxnet** [comment]
@FroggyFlox My apologies.

Re:

> Also note that in my as yet unpublished work on: (t) Failing unit tests #2371
> 
> I have a local change thus:
> 
> > ```
> > os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings")
> > ```

I'm afraid I've failed to split this one out as would have been best. I've now, for convenience, added my pending fix for what looks like a typo on my earlier part, as an unrelated part of:
(t) Services, Replication-Send_Receive, and Tasks pages fail #2370 #2380

Given you have documented here the issue we at least have that as a separate and focused explanation. I don't think it's worth doing a specific pr for this now we already have the fix in the pending pr likely about to be merged. Plus it's been quite inconvenient to have this bug in place so the sooner we get it merged the better I think.

Thanks for creating the independent issue, I'll add it with attribution to the rpm change-log.

**Comment by FroggyFlox** [comment]
> Given you have documented here the issue we at least have that as a separate and focused explanation. I don't think it's worth doing a specific pr for this now we already have the fix in the pending pr likely about to be merged. Plus it's been quite inconvenient to have this bug in place so the sooner we get it merged the better I think.

I completely agree!

Thanks for taking care of that!

**Comment by phillxnet** [comment]
Fixed by #2380
